components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/builder/WSO2LoginEventPayloadBuilder.java

```diff
@@ -17,36 +17,197 @@
+                authenticationContext.getTenantDomain()));
+        payload.setApplication(new Application(
+                authenticationContext.getServiceProviderResourceId(),
+                authenticationContext.getServiceProviderName()));
+        payload.setReason(buildAuthenticationFailedReason(authenticationContext));
+
+        return payload;
     }
 
     @Override
     public String getEventSchemaType() {
         return EVENT_SCHEMA_TYPE_WSO2;
     }
+
+    private List<String> buildAuthMethods(AuthenticationContext authContext) {
+
+        List<String> authMethods = new ArrayList<>();
+        for (AuthHistory authHistory : authContext.getAuthenticationStepHistory()) {
+            authMethods.add(authHistory.toTranslatableString());
+        }
+        return authMethods;
+    }
+
+    private AuthenticationFailedReason buildAuthenticationFailedReason(AuthenticationContext authContext) {
+
+        AuthenticationFailedReason failedReason = new AuthenticationFailedReason();
+        HashMap<String, String> dataMap = (HashMap<String, String>) authContext.getParameters().get(Constants.DATA_MAP);
+        String errorCode = dataMap.get(Constants.CURRENT_AUTHENTICATOR_ERROR_CODE);
+        failedReason.setId(errorCode);
+
+        AuthenticationFailedReason.FailedStep failedStep = new AuthenticationFailedReason.FailedStep();
+        failedStep.setStep(authContext.getCurrentStep());
+        failedStep.setAuthenticator(authContext.getCurrentAuthenticator());
+        failedStep.setIdp(authContext.getExternalIdP() != null ?
+                authContext.getExternalIdP().getIdentityProvider().getIdentityProviderName() : null);
+        failedReason.setFailedStep(failedStep);
+
+        return failedReason;
+    }
+
+    private void populateUserAttributes(AuthenticatedUser authenticatedUser, User user) throws IdentityEventException {
+
+        if (authenticatedUser == null) {
+            return;
+        }
+
+        List<UserClaim> userClaims = new ArrayList<>();
+        for (Map.Entry<ClaimMapping, String> entry : authenticatedUser.getUserAttributes().entrySet()) {
+            ClaimMapping claimMapping = entry.getKey();
+            String claimUri = claimMapping.getLocalClaim().getClaimUri();
+            String claimValue = entry.getValue();
+
+            if (claimUri != null && claimValue != null) {
+                switch (claimUri) {
+                    case Constants.WSO2_CLAIM_GROUPS:
+                        user.addGroup(claimValue);
+                        break;
+                    case Constants.WSO2_CLAIM_ROLES:
+                        user.addRole(claimValue);
+                        break;
+                    case Constants.MULTI_ATTRIBUTE_SEPARATOR:
+                        // Not adding the multi attribute separator to the user claims
+                        break;
+                    case Constants.USER_ORGANIZATION:
+                        setUserOrganization(claimValue);
```

Comment:
I think you can use the user object to set organization and utilize that when building the payload